src/Sylius/Bundle/CoreBundle/Checkout/Step/FinalizeStep.php

```diff
@@ -89,5 +89,7 @@ protected function completeOrder(OrderInterface $order)
 
         $this->dispatchCheckoutEvent(SyliusCheckoutEvents::FINALIZE_COMPLETE, $order);
         $this->dispatchCheckoutEvent(SyliusOrderEvents::POST_CREATE, $order);
+
+        $this->getCartProvider()->abandonCart();
```

Comment:
When we finalize checkout, we shouldn't have possibility to enter checkout process with the same cart, so basically we need to abandon cart.